### PR TITLE
🐛 e2e: Specify ip address pool for retreiving ip for accessing node

### LINKED ISSUE #1527: E2E feature tests are failing on heathcheck tests
### Which jobs are failing?

https://jenkins.nordix.org/job/metal3_daily_main_e2e_feature_test_ubuntu/
https://jenkins.nordix.org/job/metal3_daily_release-1-6_e2e_feature_test_ubuntu/
https://jenkins.nordix.org/job/metal3_daily_release-1-5_e2e_feature_test_ubuntu/
https://jenkins.nordix.org/job/metal3_daily_release-1-4_e2e_feature_test_ubuntu/

### Which tests are failing?

All e2e fature tests are failing on healthcheck.go  tests

### Since when has it been failing?

march 5th

### Jenkins link

https://jenkins.nordix.org/job/metal3_daily_release-1-6_e2e_feature_test_ubuntu/68/ , https://jenkins.nordix.org/job/metal3_daily_main_e2e_feature_test_ubuntu/478

### Reason for failure (if possible)

[This function](https://github.com/metal3-io/cluster-api-provider-metal3/blob/dd5cc96ce8bd05527e60aa21a26d2cd2d2ca6bc4/test/e2e/common.go#L592) is getting ip address from provisioning Ip pool, instead of external Ip pool. 
Thus [next function ](https://github.com/metal3-io/cluster-api-provider-metal3/blob/dd5cc96ce8bd05527e60aa21a26d2cd2d2ca6bc4/test/e2e/healthchek.go#L47) is failing because it cannot ssh to a node via ip address from provisioning pool.

### Anything else we need to know?

_No response_

### Label(s) to be applied

/kind failing-test
/area e2e-tests

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR makes to choose ip address from baremetal pool to access node in healthcheck tests. 
Node is not ssh able with provisioning ip address. 
Fixes #1527

### CODE REVIEW

**Review by kashifest — CHANGES_REQUESTED** [review]

File: test/e2e/common.go [hunk 1]
```diff
@@ -720,15 +720,19 @@ func (Metal3LogCollector) CollectMachineLog(ctx context.Context, cli client.Clie
 	kubeadmCP := framework.GetKubeadmControlPlaneByCluster(ctx, framework.GetKubeadmControlPlaneByClusterInput{
 		Lister:      cli,
 		ClusterName: m.Spec.ClusterName,
-		Namespace:   m.Namespace,
+		Namespace:   m.Spec.ClusterName,
```
[the remark this answers is not in the document]
Comment (Sunnatillo) [reply test/e2e/common.go]: Thanks

**Comment by Sunnatillo** [comment]
networking issue

**Comment by Sunnatillo** [comment]
/test-ubuntu-e2e-feature-main 
previous one failed with networking issue

**Review by kashifest — COMMENTED** [review]
/approve

**Comment by Sunnatillo** [comment]
Previous fail is not related to this change.
/test-ubuntu-e2e-feature-main

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Review by kashifest — APPROVED** [review]

**Pull request merged**